Login through passport

vladimyr opened the pull request:
This PR swaps custom login logic with `passport-local` strategy. Additionally notion of token's `audience` has been introduced to enable differentiating between `access` and `setup` type tokens. 🎉

vladimyr committed:
Use _raw_ axios instance for login requests

- removing JWT related intercepts for login requests
- moving api path to `.env`

vladimyr committed:
Use `passport-local` for login :tada:

vladimyr committed:
Differentiate JWT tokens by `audience`

- making distinction between `access` & `setup` (invite, reset
  password) JWT tokens

marinko-peso commented on client/common/api/request.js:
```diff
@@ -30,3 +28,6 @@ client.interceptors.response.use(res => res, err => {
 
 export default client;
+
+const base = axios.create(config);
```
Can we modify the existing axios client not to apply interceptor logic for 401 on login route instead of adding a new one?

vladimyr committed:
Expose lazy getter for base axios client

- implementing lazy getter for creation of interceptor-free axios
  instance using original configuration

marinko-peso reviewed (approved):
Looking good, nice approach with base axios instance on client axios instance 👍

kronicker commented on client/common/api/request.js:
```diff
@@ -1,13 +1,18 @@
+  get() {
+    if (!this.base_) this.base_ = axios.create(config);
+    return this.base_;
+  }
```
👌

kronicker commented on server/common/auth/index.js:
```diff
@@ -3,12 +3,26 @@
   jwtFromRequest: ExtractJwt.fromAuthHeaderWithScheme(config.scheme),
-  secretOrKey: config.secret
+  secretOrKey: config.secret,
+  audience: 'access'
```
Kudos for making JWT more specific, but I have two nerdy/nitpicky concerns:
1. Looking at the [JWT RFC](https://tools.ietf.org/html/rfc7519#section-4.1.3) I would argue that audience (`aud`) is not the best fit for the your use-case. Audience should be "the recipients that the JWT is intended for" which in our case is the same as the issuer (`iss`). We could stretch this definition by saying this is part of our API and the audince for this token but for me it seems far fetched. I think more proper way would be with out custom claim.
2. That being said, if we are to use optional JWT claims, which I'm all for, shouldn't we validate tokens based on them?

vladimyr committed:
Introduce `scope` audience namespace

- making @kronicker happy
